extensions/amp-subscriptions/0.1/local-subscription-platform.js

```diff
@@ -53,6 +56,70 @@ export class LocalSubscriptionPlatform {
         ),
         'Authorization Url'
     );
+
+    // TODO(dvoytenko, #3742): This will refer to the ampdoc once AccessService
+    // is migrated to ampdoc as well.
+    /** @private @const {!Promise<!../../../src/service/cid-impl.Cid>} */
+    this.cid_ = Services.cidForDoc(ampdoc);
+
+    /** @private {UrlBuilder} */
+    this.urlBuilder_ = null;
+
+    /** @private {Actions} */
+    this.actions_ = null;
+
+    /** @private {Promise<string>} */
+    this.readerIdPromise_ = null;
+
+    /** @private {SubscriptionAnalytics} */
+    this.subscriptionAnalytics_ = new SubscriptionAnalytics();
+
+    this.initActions_();
+  }
+
+  /**
+   * Initializes the actions for the local platform
+   * @private
+   */
+  initActions_() {
+    this.urlBuilder_ = new UrlBuilder(this.ampdoc_, this.getReaderId_());
+    user().assert(this.serviceConfig_['actions'],
+        'Actions have not been defined in the service config');
+    this.actions_ = new Actions(
+        this.ampdoc_, this.urlBuilder_,
+        this.subscriptionAnalytics_,
+        this.validateActionMap(this.serviceConfig_['actions'])
+    );
+  }
+
+  /**
+   * Validates the action map
+   * @param {!JsonObject<string, string>} actionMap
+   * @returns {!JsonObject<string, string>}
+   */
+  validateActionMap(actionMap) {
+    user().assert(actionMap['login'],
+        'Login action is not present in action map');
```

Comment:
Let's be more specific. `Action "login" must be present.....`

Reply:
Done